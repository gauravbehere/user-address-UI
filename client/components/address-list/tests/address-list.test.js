import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AddressList from '..';

describe('Test suite for AddressList component', () => {
  it('Should render AddressList component without props', () => {
    const { container, getByTestId } = render(<AddressList/>);
    expect(container).toMatchSnapshot();
    expect(getByTestId('noUserSelectedMsg')).toBeInTheDocument();
  });

  it('Should show No Address Found message if there is no address passed for a selected user', () => {
    const { getByTestId } = render(<AddressList addressList={[]} selelectedUserId={1}/>);
    expect(getByTestId('noAddressFoundMsg')).toBeInTheDocument();
  });

  it('Should show error message if there was an error fetching address', () => {
    const { getByText } = render(<AddressList addressListError='Error'/>);
    expect(getByText('Error')).toBeInTheDocument();
  });

  it('Should show render the address list', () => {
    const { getByText } = render(<AddressList addressList={[{id: 1, text: 'home 1'}]} selelectedUserId={1}/>);
    expect(getByText('home 1')).toBeInTheDocument();
  });
});
