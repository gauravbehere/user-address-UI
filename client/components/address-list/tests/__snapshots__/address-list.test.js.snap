// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for AddressList component Should render AddressList component without props 1`] = `
<div>
  <div
    class="address-list"
  >
    <div
      class="address-list-header"
    >
      Address List
    </div>
    <div
      data-testid="noUserSelectedMsg"
    >
      Pick a user
    </div>
  </div>
</div>
`;
